### DISCUSSION TITLE
Is it possible to scan a file or a folder?

### DISCUSSION THREAD

**Discussion by savushkin-yauheni** [projectdiscovery/nuclei#626] [Q&A] [answered]
Hi team!

is it possible to launch nuclei on file or folder ?

e.x. i have downloaded js files and i would like to run `exposed-tokens/generic/credentials-disclosure.yaml` template.

Thanks

**Comment by ehsandeep** [comment, marked as answer]
Hey @savushkin-yauheni, sure you can, `echo folder_name | nuclei -t exposed-tokens/generic/credentials-disclosure.yaml -passive`

**Reply by savushkin-yauheni** [reply]
Hi. Thanks for your reply. It's a great feature. Thanks

**Reply by savushkin-yauheni** [reply]
@ehsandeep Hi again! i tried and have some questions.

1. files should have `txt`  extension. until i looked at the code, i think the feature doesn't work. maybe docs should contain the note about it.

2. as i understand files should contain http raw response with headers. Am i right ? Can i bypass it ? Can i just download simple js file without headers content in it and launch nuclei ? Why i am asking. Cause i would like to run `js-beautify` on js files before nuclei launching, so `js-beautify` broke http headers and nuclei couldn't parse files after that.

Thanks a lot!

**Reply by ehsandeep** [reply]
@savushkin-yauheni looks like we are mixing two features? a) `-passive` flag works on txt file containing HTTP response, b) file protocol / template works on any file with any extension.

**Reply by savushkin-yauheni** [reply]
Hi @ehsandeep 
Yeah, you are right =)
